refgenconf/helpers.py

```diff
@@ -69,3 +80,198 @@ def get_dir_digest(path, pm=None):
+            # use the genome digest as primary keys
+            rgc[CFG_GENOMES_KEY][digest] = rgc[CFG_GENOMES_KEY].pop(genome)
+            # create "aliases" section
+            rgc[CFG_GENOMES_KEY][digest][CFG_ALIASES_KEY] = [genome]
+            # remove old "genome_digest" section
+            del rgc[CFG_GENOMES_KEY][digest][CFG_CHECKSUM_KEY]
+        else:
+            del rgc[CFG_GENOMES_KEY][genome]
+
+
+def alter_file_tree_03_04(rgc, link_fun):
+    """
+    update file structure inside genome_folder:
+    Drop genomes for which genome_digest is not available
+    on any of the servers and do not have a fasta asset locally.
+    contents inside genome_folder will be replaced by 'alias' and 'data' dir
+
+    :param obj rgc: RefGenConfV03 obj
+    :param callable link_fun: function to use to link files, e.g os.symlink
+        or os.link
+    """
+    my_genome = {}
+    for k, v in rgc[CFG_GENOMES_KEY].items():
+        my_genome.update([(v[CFG_ALIASES_KEY][0], k)])
+
+    _LOGGER.info(f"Creating '{DATA_DIR}' and '{ALIAS_DIR}' directories in "
+                 f"'{rgc[CFG_FOLDER_KEY]}'.")
+    os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)))
+    os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR)))
+
+    _LOGGER.info(
+        f"Copying assets to '{DATA_DIR}' and creating alias symlinks in "
+        f"'{ALIAS_DIR}'. Genomes that the digest could not be determined for "
+        f"will be ignored.")
+    for root, dirs, files in os.walk(rgc[CFG_FOLDER_KEY]):
+        for dir in dirs:
+            if dir in my_genome:
+                shutil.copytree(os.path.join(rgc[CFG_FOLDER_KEY], dir),
+                                os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR,
+                                             dir), symlinks=True)
+        del dirs[:]
+
+    for root, dirs, files in os.walk(
+            os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)):
+        for dir in dirs:
+            swap_names_in_tree(os.path.join(root, dir), my_genome[dir], dir)
+            os.mkdir(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR, dir))
+            # create symlink for alias folder
+            for genome, assets, files in os.walk(
+                    os.path.join(root, my_genome[dir])):
+                for asset in assets:
+                    old_path = os.path.join(genome, asset)
+                    new_path = old_path.replace(my_genome[dir], dir).replace(
+                        DATA_DIR, ALIAS_DIR)
+                    os.mkdir(new_path)
+                for file in files:
+                    des_path = os.path.join(genome, file)
+                    src_path = os.path.realpath(des_path).replace(rgc[CFG_FOLDER_KEY], 
+                                os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)).replace(dir, my_genome[dir])
+                    
+                    if os.path.islink(des_path):
+                        os.remove(des_path)
+                        link_fun(src_path, des_path)
```

Comment:
since `link_fun` is an argument to this function, the effect of its execution can be anything; you can't assume that it will create a symlink. If a file was a symlink in the original dir, it should remain one.